src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs

```diff
@@ -132,21 +132,16 @@ protected virtual void CheckLabelCompatible(SchemaShape.Column labelCol)
         protected TTransformer TrainTransformer(IDataView trainSet,
             IDataView validationSet = null, IPredictor initPredictor = null)
         {
-            var cachedTrain = Info.WantCaching ? new CacheDataView(Host, trainSet, prefetch: null) : trainSet;
```

Comment:
As requested by @GalOshri in the issue, can we [add documentation](https://github.com/dotnet/machinelearning/issues/1604#issuecomment-440824201)? 

Currently the user will have no method of knowing if a specific learner already does it own form of caching, or won't benefit from caching.

Inline w/ @GalOshri's request, I think this documentation should be required before making this change.

Reply:
Let's change the appropriate cookbook samples to illustrate the new pattern with this little caching checkpoint thing.

---
In reply to: [237951473](https://github.com/dotnet/machinelearning/pull/1780#discussion_r237951473) [](ancestors = 237951473)

Reply:
Yes, updating the example code is a good first step. And we should create a direct list of the components which benefit from caching. This is along with _when_ they benefit, for instance, "a LinearSVM when the number of iterations are greater than 1".

Another route is perhaps a VS checker which look at Info.WantCaching and recommends from there?

Reply:
A sample and some tests are modified to use those caching functions. Every caching function has at least one test now.

Reply:
I don't think having a list is a small task. We need another PR.

---
In reply to: [237953879](https://github.com/dotnet/machinelearning/pull/1780#discussion_r237953879) [](ancestors = 237953879)

Reply:
Ok. I will do it in next iteration.

---
In reply to: [237951764](https://github.com/dotnet/machinelearning/pull/1780#discussion_r237951764) [](ancestors = 237951764,237951473)

Reply:
Generally, I like to see documentation in the PR. This is more so true when the user can be surprised by the change and not understand what's different.

Reply:
Many uses are added into Coookbook.

---
In reply to: [238949665](https://github.com/dotnet/machinelearning/pull/1780#discussion_r238949665) [](ancestors = 238949665)